airflow-core/src/airflow/ui/src/pages/XCom/XCom.tsx

```diff
@@ -82,18 +83,24 @@ const columns = (translate: (key: string) => string, open: boolean): Array<Colum
             taskId: original.task_id,
           })}
         >
-          <TruncatedText text={original.task_id} />
+          <TruncatedText text={original.task_display_name || original.task_id} />
```

Comment:
Nit:

Task display name can't be None or undefined, we probably should just show this. ("empty string will default to `task_id` in your case but I don't think it's important.)

Also this is more consistent with task instances and HITL tables.
```suggestion
          <TruncatedText text={original.task_display_name} />
```

Reply:
@pierrejeambrun  Thanks for the clarification and the approval! I've removed the redundant fallback logic as suggested and rebased the branch. Ready for merge!